File: python/perfect.py
import math

def int_square_root(value: int) -> int:
    """Calculates the integer square of an integer.

       Attempted to write on function using Newton's method however
       it ran at half the speed of this metod."""
    return int(math.sqrt(value))

def is_perfect_square(n: int) -> int:
    try:
        root: int = int_square_root(n)
        return root*root == n
    except ValueError:
        return False

File: c++/cpp/perfect.cpp
#include <cmath>
#include <stdexcept>

int intSquareRoot(int value)
{
    // Calculates the integer square root of an integer.
    if (value < 0)
    {
        throw std::out_of_range("Out of range");
    }
    return int(sqrt(value));
}

int isPerfectSquare(int value)
{
    int root = intSquareRoot(value);
    return (root * root == value);
}
